modules/memberships/class-jetpack-memberships.php

```diff
@@ -227,21 +227,81 @@ public function allow_sync_post_meta( $post_meta ) {
 	public function return_meta( $map ) {
 		return $map['meta'];
 	}
+
 	/**
-	 * Callback that parses the membership purchase shortcode.
+	 * Renders a preview of the Recurring Payment button, which is not hooked
+	 * up to the subscription url. Used to preview the block on the frontend
+	 * for admin users when Stripe has not been connected.
 	 *
-	 * @param array  $attrs - attributes in the shortcode. `id` here is the CPT id of the plan.
+	 * @param array  $attrs - attributes in the shortcode.
 	 * @param string $content - Recurring Payment block content.
 	 *
 	 * @return string|void
 	 */
-	public function render_button( $attrs, $content = null ) {
+	public function render_button_preview( $attrs, $content = null ) {
+		if ( ! empty( $content ) ) {
+			$block_id = esc_attr( wp_unique_id( 'recurring-payments-block-' ) );
+			$content  = str_replace( 'recurring-payments-id', $block_id, $content );
+			return $content;
+		}
+		return $this->deprecated_render_button_v1( $attrs, null );
+	}
+
+	/**
+	 * Determines whether the button preview should be rendered. Returns true
+	 * if the user is an admin, the button is not configured correctly
+	 * (because it requires a plan upgrade or Stripe connection), and the
+	 * button is a child of a Premium Content block.
+	 *
+	 * @param WP_Block $block Recurring Payments block instance.
+	 *
+	 * @return boolean
+	 */
+	public function should_render_button_preview( $block ) {
+		$is_admin                   = $this->user_can_edit();
+		$requires_stripe_connection = ! $this->get_connected_account_id();
+
+		$requires_upgrade = false;
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM && function_exists( 'has_any_blog_stickers' ) ) {
+			$site_id          = get_current_blog_id();
+			$requires_upgrade = has_any_blog_stickers( array( 'personal-plan', 'premium-plan', 'business-plan', 'ecommerce-plan' ), $site_id );
+		} else {
+			$requires_upgrade = Jetpack::is_active() && Jetpack_Plan::supports( 'recurring-payments' );
```

Comment:
Do we need to check for the `jetpack_block_editor_enable_upgrade_nudge` filter here, like we do elsewhere?

Reply:
No, we need to exclude that check here -- which is also unfortunately why we can't just use `is_enabled_jetpack_recurring_payments` for the check.

For the purpose of displaying the block preview, we need to check whether the site requires an upgrade in order to fully support the block. The filter indicates that the block should be enabled with the upgrade nudge if is not supported. It is used to determine whether the block should be registered.